contract/src/main.rs

```diff
@@ -0,0 +1,138 @@
+#![no_main]
+#![allow(unused_imports)]
+#![allow(unused_parens)]
+#![allow(non_snake_case)]
+
+extern crate alloc;
+
+use alloc::{
+    collections::{BTreeMap, BTreeSet},
+    string::String,
+};
+use core::convert::TryInto;
+
+use casperlabs_contract::{
+    contract_api::{runtime, storage},
+    unwrap_or_revert::UnwrapOrRevert,
+};
+use casperlabs_contract_macro::{casperlabs_constructor, casperlabs_contract, casperlabs_method};
+use casperlabs_types::{
+    account::AccountHash,
+    bytesrepr::{FromBytes, ToBytes},
+    contracts::{EntryPoint, EntryPointAccess, EntryPointType, EntryPoints},
+    runtime_args, CLType, CLTyped, CLValue, Group, Parameter, RuntimeArgs, URef, U256,
+};
+
+#[casperlabs_contract]
+mod ERC20 {
+
+    #[casperlabs_constructor]
+    fn constructor(tokenName: String, tokenSymbol2: String, tokenTotalSupply: U256) {
+        let _name: String = tokenName;
+        set_key("_name", _name);
+        let _symbol: String = tokenSymbol2;
+        set_key("_symbol", _symbol);
+        let _decimals: u8 = 18;
+        set_key("_decimals", _decimals);
+        let temp8: U256 = tokenTotalSupply;
+        set_key(&new_key("_balances", runtime::get_caller()), temp8);
+        let _totalSupply: U256 = tokenTotalSupply;
+        set_key("_totalSupply", _totalSupply);
+    }
+
+    #[casperlabs_method]
+    fn name() {
+        ret(get_key::<String>("_name"));
```

Comment:
@darthsiroftardis haven't you implemented `returns` in DLS?

Reply:
I did, it pointed to a previous version